src/ServiceControl.Transports/IProvideQueueLength.cs

```diff
@@ -1,16 +1,9 @@
 ﻿namespace ServiceControl.Transports
 {
-    using System;
-    using System.Threading.Tasks;
+    using Microsoft.Extensions.Hosting;
 
-    public interface IProvideQueueLength
+    public interface IProvideQueueLength : IHostedService
```

Comment:
I feel that the implementers need to decide whether using `IHostedService` is the right thing to do.
in other words, should this interface **not** implement `IHostedService `?

Reply:
That's an excellent question. I was going down that path because the existing code and test was starting and stopping things to make it work. By not adding the hosted service interface the test code would become slightly more complex because it would have to retrieve the `AbstractQueueLengthProvider` and then start that. This would also mean we then technically need three registrations of which one would only ever be used by the transport tests. We can't just retrieve the `IHostedService` instance because while there is currently only one, there might be more later. So I'm kind of torn. I'll merge the PR and think about this a bit more.

Reply:
Yeah, I can see the complexity.
TBH, given this is not a public API, I am fine with it 😄